# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1246277759 -7200
#      Mon Jun 29 14:15:59 2009 +0200
# Branch release-3-2-x
# Node ID 9d85d65615731b1ffe94926f56846dd31c264284
# Parent  6d6e703f467c7620daf5efce3f5b2960916a9f36
load-path.cc (Faddpath): preserve order of prepended elements

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-06-26  John W. Eaton  <jwe@octave.org>
+
+	* load-path.cc (Faddpath): Preserve order of prepended elements.
+
 2009-06-24  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* oct-map.cc (Octave_map::squeeze, Octave_map::permute,
 	Octave_map::transpose, Octave_map::reshape, Octave_map::concat,
 	Octave_map::index): Add tests for preservation of key order in
 	struct arrays.
 
 2009-06-24  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/src/load-path.cc b/src/load-path.cc
--- a/src/load-path.cc
+++ b/src/load-path.cc
@@ -2019,16 +2019,19 @@ Directories added to the path must exist
       for (int i = 0; i < nargin; i++)
 	{
 	  std::string arg = args(i).string_value ();
 
 	  if (! error_state)
 	    {
 	      std::list<std::string> dir_elts = split_path (arg);
 
+	      if (! append)
+		std::reverse (dir_elts.begin (), dir_elts.end ());
+
 	      for (std::list<std::string>::const_iterator p = dir_elts.begin ();
 		   p != dir_elts.end ();
 		   p++)
 		{
 		  std::string dir = *p;
 
 		  //dir = regexprep (dir_elts{j}, "//+", "/");
 		  //dir = regexprep (dir, "/$", "");
